nucliadb/tests/nucliadb/integration/test_notifications.py

```diff
@@ -0,0 +1,172 @@
+# Copyright (C) 2021 Bosutech XXI S.L.
+#
+# nucliadb is offered under the AGPL v3.0 and as commercial software.
+# For commercial licensing, contact us at info@nuclia.com.
+#
+# AGPL:
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+import asyncio
+from typing import Union
+from unittest.mock import patch
+
+import httpx
+import pytest
+import uvicorn
+from httpx import AsyncClient
+
+from nucliadb.reader.api.v1.router import KB_PREFIX
+from nucliadb.reader.app import create_application
+from nucliadb_models.notifications import (
+    Notification,
+    ResourceIndexedNotification,
+    ResourceProcessedNotification,
+    ResourceWrittenNotification,
+)
+from nucliadb_protos.writer_pb2 import BrokerMessage
+from nucliadb_utils.settings import transaction_settings
+from tests.ingest.fixtures import broker_resource
+
+
+# This is a clone of `reader_api_server` but using nidx instead of dummy_nidx_utility
+@pytest.fixture(scope="function")
+async def nidx_reader_api_server(
+    storage,
+    maindb_driver,
+    local_files,
+    indexing_utility,
+    nidx,
+):
+    application = create_application()
+    with patch.object(transaction_settings, "transaction_local", True):
+        async with application.router.lifespan_context(application):
+            yield application
```

Comment:
Maybe you could, instead of redefining a "private" fixture, call the original and then pass a nidx_utility fixture which overwrites the utility.

Reply:
I tried it at some point and it didn't work, due to the order of initialization of several fixtures. Since then,. I removed a bunch of them (e.g: nucliadb_writer), so let's try it again.

Reply:
Okay, it worked fine, thanks for the tip!

I also moved this to the search directory since it uses the nidx image and it's only built in CI in that environment. Not the cleanest, but I don't think that's horrible, is it?